### PR TITLE
POC - MGDOBR 404

### PR DESCRIPTION
**Pull request by lucamolteni**
[MGDOBR-404](https://issues.redhat.com/browse/MGDOBR-404)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [ ] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [ ] The layers in the `kustomize` folder have been updated accordingly.
- [ ] All new functionality is tested
- [ ] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: dev/bin/interaction/create-action.sh [hunk 1]
```diff
@@ -57,8 +57,8 @@ elif [ "${action_type}" = 'slack-plain' ]; then
    "action": {
       "type": "Slack",
       "parameters": {
-         "channel": "mc",
-         "webhookUrl": '"\"$SLACK_WEBHOOK_URL\""'
+         "slack_channel": "mc",
+         "slack_webhook_url": '"\"$SLACK_WEBHOOK_URL\""'
       }
```
Comment (r00ta) [inline dev/bin/interaction/create-action.sh:62]: Just brainstorming: we could change the `type` to `Connector` and provide the connector type inside the `parameters`. In this way we can better separate what is a connector and what is not.

File: processor/configuration/src/main/java/com/redhat/service/smartevents/processor/actions/slack/SlackConnectorAction.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.redhat.service.smartevents.processor.actions.slack;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import com.redhat.service.smartevents.processor.actions.generic.GenericConnectorAction;
+
+@ApplicationScoped
+public class SlackConnectorAction extends GenericConnectorAction {
```
Comment (r00ta) [inline processor/configuration/src/main/java/com/redhat/service/smartevents/processor/actions/slack/SlackConnectorAction.java:8]: Brainstorming: we could remove all the connector specific classes with the following workflow: 
1) We get a connector action
2) if the `type` is for example `slack_sink_0.1`, we look for the `slack_sink_0.1.json` file
3) we use `slack_sink_0.1.json` for the validation

In this way we can
1) add a new connector by just adding the json in the `resources` folder
2) no need to write a single line of code for any connector

On the other side, if we keep connector-specific classes we might add additional checks on the payload. For example if a parameter is an URL, the json schema validation would just check that it is a string, but we could also check that the value is a valid URL.

**Review by r00ta — COMMENTED** [review]
Thank you for starting with this, I think it will improve a lot the way we will manage the integration with connectors

**Comment by vaibhavjainwiz** [comment]
@Brainstorming : Currently we have our own contract which we expose to external user. Which provide us full control how we want to provide support for it in future. For example In future if we have to capture some extra field for Slack action then we could add new parameter.
```
"name": "$action_name"',
   "action": {
      "type": "Slack",
      "parameters": {
         "channel": "mc",
         "webhookUrl": '"$SLACK_WEBHOOK_URL"'
      }
   }
```
On other hand if we choose to use MC schema then we can't add/remove field on bases of our requirement. It will also tightly bind RHOSE with MC only. 
**Cons : **

- If our customer requirs to connect with an application whose MC is not yet implemented then we can't write custome connect logic for it.
- If MC choose to add some new paramter or rename existing paramter then our application users start getting error.

**Pull request closed without merging**